### PR TITLE
Bug 1291173 - Renamed memory measures keys to have underscores.

### PR DESCRIPTION
**Pull request by adngdb**
These keys had dashes in their names, leading to inconsistent naming accross our crash documents. That was to make extracting values from the memory report simpler, but we should not sacrifice naming conventions. With this all keys use underscore naming.

cc @nnethercote

### CODE REVIEW

**Comment by willkg** [comment]
This seems fine. I went through and double-checked that we didn't have other rules or something that were looking at the old key names and didn't see anything obvious.

r+

File: socorro/processor/rules/memory_report_extraction.py [hunk 1]
```diff
@@ -82,15 +86,21 @@ def _get_memory_measures(self, memory_report, pid):
...
             'images': 0,
-            'js-main-runtime': 0,
-            'top-non-detached': 0,
+            'js_main_runtime': 0,
+            'top_non_detached': 0,
         }
 
         all_metrics = {}
-        all_metrics.update(metrics_measured)
+        # Replace the dashes in keys with underscores for consistency.
```
Comment (nnethercote) [inline socorro/processor/rules/memory_report_extraction.py]: Just do this step at the end, and then all the other changes to this script won't be necessary. That way the script can uses dashes all throughout, which fits with how memory reporting works, and then we switch to underscores in a single place.

Comment (adngdb) [inline socorro/processor/rules/memory_report_extraction.py]: :+1:

**Comment by adngdb** [comment]
@nnethercote I've changed the code as you suggested, it's much simpler. And I also renamed ``top-non-detached`` to ``top-none-detached``. r?

**Comment by nnethercote** [comment]
Much better! Can you please fold the two patches together before landing?

**Pull request merged**